lib/sbuf.h

```diff
@@ -0,0 +1,20 @@
+#ifndef SBUF_H
```

Comment:
Can we also get added some documentation on how this is meant to be used added directly to this .h as well?